tests/psa-client-server/psasim/include/util.h

```diff
@@ -13,20 +13,18 @@
 #if defined(DEBUG)
 #define INFO(fmt, ...) \
     fprintf(stdout, "Info (%s - %d): " fmt "\n", __FILE__, __LINE__, ##__VA_ARGS__)
+#else /* DEBUG */
+#define INFO(...)
+#endif /* DEBUG*/
 
 #define ERROR(fmt, ...) \
-    fprintf(stdout, "Error (%s - %d): " fmt "\n", __FILE__, __LINE__, ##__VA_ARGS__)
+    fprintf(stderr, "Error (%s - %d): " fmt "\n", __FILE__, __LINE__, ##__VA_ARGS__)
 
 #define FATAL(fmt, ...) \
     { \
-        fprintf(stdout, "Fatal (%s - %d): " fmt "\n", __FILE__, __LINE__, ##__VA_ARGS__); \
+        fprintf(stderr, "Fatal (%s - %d): " fmt "\n", __FILE__, __LINE__, ##__VA_ARGS__); \
         abort(); \
```

Comment:
Not a blocker but a design question.

`FATAL` aborts the execution flow, while `ERROR` is just reporting an error.

Should we add a FAIL macro which does 

    #define FAIL(fmt, ...) \
        fprintf(stderr, "Error (%s - %d): " fmt "\n", __FILE__, __LINE__, ##__VA_ARGS__)
        return -1

And use it bellow?

Reply:
> Should we add a FAIL macro which does

There is not an uniform `return` value used in `psasim` so far. Some functions returns error with negative values, while other return 0 as failure. I'm not sure we would use this macro that much.

> And use it bellow?

Where exactly?

Reply:
I think we should at least try to be consistent on the code that we are adding when it comes to using those new utility macros.

My take is:

- We should not be failing something and return 0
- Failure retcode can be -1, 1 or anything other than zero
- If we need to check for failure we expect it to be not zero.


> Where exactly?

In the psa_serialise.c there are cases of `ERROR` followed by `return -1` .

Either was that is a non blocker discussion that we should consider moving forwards.

Reply:
> Failure retcode can be -1, 1 or anything other than zero

I agree, but currently serializing/deserializing code uses the opposite pattern (0 for errors, 1 for success) so we need to update that part and the generating Perl script as well. I think this deserves a specific PR.

Reply:
Yeah, that's a consequence of me developing those in my free time. There is allowance for "success is truthy" in the Mbed TLS coding style, of course, but we don't often do it now